### PR TITLE
Atomic/move

### PR DESCRIPTION
**Pull request by lpoli**
### Changes
This PR had made move operation atomic. Previously it would be copy + delete in which client was charged for copy operation.

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber: https://github.com/0chain/blobber/pull/866
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | atomic/move  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | atomic/move  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | atomic/move  |
| zboxcli  | atomic/move  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | atomic/move  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: zboxcore/allocationchange/moveobject.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+				Name:         fields[i],
+			}
+			dirRef.AddChild(newRef)
+			dirRef = newRef
+		}
+		dirRef.HashToBeComputed = true
+	}
+
+	if dirRef.GetPath() != ch.DestPath || dirRef.GetType() != fileref.DIRECTORY {
+		return errors.New("file_not_found", "Object to m9ve not found in blobber")
```
Comment (peterlimg) [inline zboxcore/allocationchange/moveobject.go]: m9ve -> move

File: zboxcore/sdk/moveworker.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	numList := len(req.blobbers)
+	objectTreeRefs := make([]fileref.RefEntity, numList)
+	wg := &sync.WaitGroup{}
+
+	var pos uint64
+
+	for i := req.moveMask; !i.Equals64(0); i = i.And(zboxutil.NewUint128(1).Lsh(pos).Not()) {
+		pos = uint64(i.TrailingZeros())
+		wg.Add(1)
+		go func(blobberIdx int) {
```
Comment (peterlimg) [inline zboxcore/sdk/moveworker.go:155]: call  defer wg.Done()  here instead of  passing it to the moveBlobberObject

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | atomic/move  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | atomic/move  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | atomic/move  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**